### PR TITLE
feat: Add MLflow Tracking blueprint

### PR DESCRIPTION
**Pull request by ovaleanu**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/awslabs/data-on-eks/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

This PR adds a blueprints for deploying MLFlow Tracking on Amazon EKS. 
MLflow Tracking is an API and UI for logging parameters, code versions, metrics, and artifacts when running your machine learning code and for later visualizing the results. 

### Motivation

Resolve [#16](https://github.com/awslabs/data-on-eks/issues/16) 

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Mandatory for new blueprints. Yes, I have added a example to support my blueprint PR
- [ ] Mandatory for new blueprints. Yes, I have updated the `website/docs` or `website/blog` section for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR. Link for installing [pre-commit](https://pre-commit.com/) locally

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

File: ai-ml/mlflow/versions.tf [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      source  = "gavinbunney/kubectl"
+      version = ">= 1.14"
+    }
+  }
+
+  # ##  Used for end-to-end testing on project; update to suit your needs
+  # backend "s3" {
+  #   bucket = "doeks-github-actions-e2e-test-state"
+  #   region = "us-west-2"
+  #   key    = "e2e/argo-workflow/terraform.tfstate"
```
Comment (vara-bonthu) [inline ai-ml/mlflow/versions.tf:31]: change `argo-workflow` to `mflow`

File: ai-ml/mlflow/addons.tf [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+  enable_ingress_nginx = true
+  ingress_nginx = {
+    values = [templatefile("${path.module}/helm-values/ingress-nginx-values.yaml", {})]
+  }
+
+  helm_releases = {
+    #---------------------------------------
+    # NVIDIA Device Plugin Add-on
+    #---------------------------------------
+    nvidia-device-plugin = {
```
Comment (vara-bonthu) [inline ai-ml/mlflow/addons.tf:135]: Use Data Addons module for this add-on

File: ai-ml/mlflow/main.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+provider "kubectl" {
+  apply_retry_count      = 10
+  host                   = module.eks.cluster_endpoint
+  cluster_ca_certificate = base64decode(module.eks.cluster_certificate_authority_data)
+  load_config_file       = false
+  token                  = data.aws_eks_cluster_auth.this.token
+}
+
+data "aws_availability_zones" "available" {}
+# data "aws_region" "current" {}
```
Comment (vara-bonthu) [inline ai-ml/mlflow/main.tf:35]: remove this commented line

**Review by vara-bonthu — COMMENTED** [review]
LGTM. I left few minor comments

**Review by askulkarni2 — APPROVED** [review]

**Pull request merged**